app/src/main/java/de/danoeh/antennapod/fragment/FeedItemlistFragment.java

```diff
@@ -106,6 +115,32 @@
     private View header;
     private Toolbar toolbar;
     private ToolbarIconTintManager iconTintManager;
+    // fab speed dial
+
+    /**
+     * Specify an action (defined by #flag) 's UI bindings.
+     *
+     * Includes: the menu / action item and the actual logic
+     */
+    public static class ActionBinding {
```

Comment:
I've studied this part in more detail and I've concluded that this is really just unneeded complexity.  Whoever wrote this was trying to implement good architecture. And while this approach would work in another project, it doesn't work for ours. It deviates from the rest of the code that only requires a switch statement. I say we make it consistent and replace this with a switch statement just like the rest of the code.

Reply:
I think it makes sense to have it as a different class. Otherwise, the same `switch` has to be copied to other screens like queue, downloads, etc.

This abstraction also makes it easier to test the code.

Reply:
Well, it don't undestand why this requires testing. It is just a common switch statment for handinling menu click events. 
The code in the click listenrs is just a database call and a call to show a message and exit the multi select mode. Not considering strictly UI related code, it is basically only a database call which we likely already have tests for. 

About other the other screens. I suggest we use a base fragment. That way they can share common logic like the switch.

Reply:
> About other the other screens. I suggest we use a base fragment. That way they can share common logic like the switch.

That's exactly why I think that a class for the ActionBinding is useful. When pushing everything to a parent fragment, future developers will tend to add more and more code to that parent fragment, making it some kind of "god class".

> Well, it don't undestand why this requires testing.

For screens with dynamic loading, the adapter does not know about all items. When someone used "select all", we have to apply the change even to items that are not loaded yet. This is currently not a problem because the screens that support dynamic loading simply don't use multi-select but for better performance, this will be needed in the future. It's not the scope of this PR but it will make future changes easier.

Reply:
> That's exactly why I think that a class for the ActionBinding is useful. When pushing everything to a parent fragment, future developers will tend to add more and more code to that parent fragment, making it some kind of "god class".

Ok, then i'll create the Actioninding class.

Reply:
> For screens with dynamic loading, the adapter does not know about all items. When someone used "select all", we have to apply the change even to items that are not loaded yet. This is currently not a problem because the screens that support dynamic loading simply don't use multi-select but for better performance, this will be needed in the future. It's not the scope of this PR but it will make future changes easier.

I'm not sure what you mean. How does this relate to testing.

Reply:
Sorry. I meant that the features will likely need to be enhanced and will get more complex. Then testing is useful.

Reply:
@ByteHamster Oh right. So how should I proceed with this PR?